java-symbolic-execution/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -910,7 +908,12 @@ private void enqueueExceptionalPaths(ProgramState ps, Symbol methodSymbol, @Null
         .orElse(exitBlock);
       enqueue(new ProgramPoint(methodExit), ps, true, methodYield);
     } else {
-      otherBlocks.forEach(b -> enqueue(new ProgramPoint(b), ps, true, methodYield));
+      var exception = ps.peekValue();
```

Comment:
Why do we peek a value from the stack instead of using the above `exceptionSV` variable?
And why not changing the way we initialize the above `exceptionSV` by using `.unstackValue(1)` instead of `peekVallue()`, and then always initialize `withEntryException` at the beginning of this method to benefit to following program states?

I would also not only set `withEntryException` in this `enqueueExceptionalPaths` method but also in `ExplodedGraphWalker#handleBlockExit` / `case THROW_STATEMENT:`

Reply:
So I improved it a little. Unfortunately, I can't change how exceptions are handled in more parts of the engine without causing issues with flows. My proposal here would be to leave it as it is for now and in future do a sprint on proper stack and exception handling, where we can git rid of some of the hacks that are currently deeply-embedded in the engine.